File: scripts/validate-config.py
#!/usr/bin/env python3
"""
APEX System Configuration Validator - Python Version
Validates that all required environment variables are properly configured
Run this before starting Python components to ensure proper setup
"""

import sys
import os

# Add src/python to path for imports
sys.path.insert(0, os.path.join(os.path.dirname(__file__), '..', 'src', 'python'))

from config import (
    validate_config,
    get_config_summary,
    get_mode_display,
    CURRENT_MODE,
    ExecutionMode
)


def print_header():
    """Print header"""
    print("\n" + "=" * 60)
    print("APEX SYSTEM CONFIGURATION VALIDATOR - PYTHON")
    print("=" * 60 + "\n")


def print_summary(config):
    """Print configuration summary"""
    print(get_mode_display())
    print()
    
    print("Configuration Summary:")
    print("─" * 60 + "\n")
    
    # Display execution mode
    print("Execution Mode:")
    print(f"  {config['mode']}\n")
    
    # Display chain configuration
    print("Configured Chains:")
    for chain_name, chain_config in config['chains'].items():
        status = "✓" if chain_config['rpc_url'] else "✗"
        print(f"  {status} {chain_config['name']} ({chain_config['chain_id']})")
    print()
    
    # Display safety parameters
    print("Safety Parameters:")
    print(f"  Min Profit: ${config['safety']['min_profit_usd']}")
    print(f"  Max Gas Price: {config['safety']['max_gas_price_gwei']} Gwei")
    print(f"  Slippage: {config['safety']['slippage_bps']} bps ({config['safety']['slippage_bps'] / 100:.2f}%)")
    print(f"  Max Daily Loss: ${config['safety']['max_daily_loss_usd']}")
    print()
    
    # Display ML configuration
    print("ML Configuration:")
    print(f"  Confidence Threshold: {config['ml']['confidence_threshold']}")
    print(f"  ML Filtering: {'Enabled' if config['ml']['enable_filtering'] else 'Disabled'}")
    print()
    
    # Display advanced features
    print("Advanced Features:")
    print(f"  Cross-Chain: {'Enabled' if config['system']['enable_cross_chain'] else 'Disabled'}")
    print(f"  Mempool Monitoring: {'Enabled' if config['system']['enable_mempool_monitoring'] else 'Disabled'}")
    print(f"  Rust Engine: {'Enabled' if config['system']['rust_engine_enabled'] else 'Disabled'}")
    print(f"  BloXroute: {'Enabled' if config['bloxroute']['enabled'] else 'Disabled'}")
    print(f"  Telegram Alerts: {'Enabled' if config['telegram']['enabled'] else 'Disabled'}")
    print()
    
    # Display execution configuration
    print("Execution Configuration:")
    print(f"  Execute Real Transactions: {'YES' if config['execution']['execute_transactions'] else 'NO (Dry-run)'}")
    print(f"  Simulate Transactions: {'YES' if config['execution']['simulate_transactions'] else 'NO'}")
    print()
    
    # Display warnings
    if config['mode'] == ExecutionMode.LIVE.value:
        print("⚠️  WARNING: LIVE MODE ENABLED")
        print("   This will execute REAL transactions on-chain")
        print("   Ensure you have tested thoroughly in DEV mode first!\n")
    
    if not config['wallet']['private_key']:
        print("⚠️  No private key configured")
        print("   Required for LIVE mode, optional for DEV/SIM\n")


def main():
    """Main validation function"""
    print_header()
    
    try:
        # Run validation
        validate_config()
        
        print("✅ Configuration validation passed!\n")
        
        # Get and display configuration summary
        config = get_config_summary()
        print_summary(config)
        
        print("✅ Python components are ready to start!\n")
        return 0
        
    except ValueError as e:
        print("❌ Configuration validation failed!\n")
        print(str(e))
        print()
        print("Please check your .env file and ensure all required variables are set.")
        print("Refer to .env.example for configuration template.\n")
        return 1
    except Exception as e:
        print(f"❌ Unexpected error during validation: {e}\n")
        return 1


if __name__ == '__main__':
    sys.exit(main())

File: scripts/validate-config.js
#!/usr/bin/env node

/**
 * APEX System Configuration Validator
 * Validates that all required environment variables are properly configured
 * Run this before starting the system to ensure proper setup
 */

import chalk from 'chalk';
import { 
    validateConfig,
    getConfig,
    getModeDisplay,
    CURRENT_MODE,
    MODE
} from '../src/utils/config.js';

console.log(chalk.bold.cyan('\n========================================'));
console.log(chalk.bold.cyan('APEX SYSTEM CONFIGURATION VALIDATOR'));
console.log(chalk.bold.cyan('========================================\n'));

console.log(getModeDisplay());
console.log();

// Run validation
try {
    const isValid = validateConfig();
    
    if (isValid) {
        console.log(chalk.green('✅ Configuration validation passed!\n'));
        
        // Display configuration summary (with sensitive data redacted)
        console.log(chalk.cyan('Configuration Summary:'));
        console.log(chalk.gray('─────────────────────────────────────────\n'));
        
        const config = getConfig();
        
        // Display execution mode
        console.log(chalk.yellow('Execution Mode:'));
        console.log(`  ${config.mode}\n`);
        
        // Display chain configuration
        console.log(chalk.yellow('Configured Chains:'));
        Object.entries(config.chains).forEach(([key, chain]) => {
            const status = chain.rpcUrl ? chalk.green('✓') : chalk.red('✗');
            console.log(`  ${status} ${chain.name} (${chain.chainId})`);
        });
        console.log();
        
        // Display safety parameters
        console.log(chalk.yellow('Safety Parameters:'));
        console.log(`  Min Profit: $${config.safety.minProfitUSD}`);
        console.log(`  Max Gas Price: ${config.safety.maxGasPriceGwei} Gwei`);
        console.log(`  Slippage: ${config.safety.slippageBps} bps (${(config.safety.slippageBps / 100).toFixed(2)}%)`);
        console.log(`  Max Daily Loss: $${config.safety.maxDailyLossUSD}`);
        console.log();
        
        // Display ML configuration
        console.log(chalk.yellow('ML Configuration:'));
        console.log(`  Confidence Threshold: ${config.ml.confidenceThreshold}`);
        console.log(`  ML Filtering: ${config.ml.enableFiltering ? chalk.green('Enabled') : chalk.gray('Disabled')}`);
        console.log();
        
        // Display advanced features
        console.log(chalk.yellow('Advanced Features:'));
        console.log(`  Cross-Chain: ${config.system.enableCrossChain ? chalk.green('Enabled') : chalk.gray('Disabled')}`);
        console.log(`  Mempool Monitoring: ${config.system.enableMempoolMonitoring ? chalk.green('Enabled') : chalk.gray('Disabled')}`);
        console.log(`  Rust Engine: ${config.system.rustEngineEnabled ? chalk.green('Enabled') : chalk.gray('Disabled')}`);
        console.log(`  BloXroute: ${config.bloxroute.enabled ? chalk.green('Enabled') : chalk.gray('Disabled')}`);
        console.log(`  Telegram Alerts: ${config.telegram.enabled ? chalk.green('Enabled') : chalk.gray('Disabled')}`);
        console.log();
        
        // Display execution configuration
        console.log(chalk.yellow('Execution Configuration:'));
        console.log(`  Execute Real Transactions: ${config.execution.executeTransactions ? chalk.red('YES') : chalk.green('NO (Dry-run)')}`);
        console.log(`  Simulate Transactions: ${config.execution.simulateTransactions ? chalk.green('YES') : chalk.gray('NO')}`);
        console.log();
        
        // Display warnings
        if (config.mode === MODE.LIVE) {
            console.log(chalk.red.bold('⚠️  WARNING: LIVE MODE ENABLED'));
            console.log(chalk.red('   This will execute REAL transactions on-chain'));
            console.log(chalk.red('   Ensure you have tested thoroughly in DEV mode first!\n'));
        }
        
        if (!config.wallet.privateKey) {
            console.log(chalk.yellow('⚠️  No private key configured'));
            console.log(chalk.gray('   Required for LIVE mode, optional for DEV/SIM\n'));
        }
        
        console.log(chalk.green('✅ System is ready to start!\n'));
        process.exit(0);
    }
} catch (error) {
    console.log(chalk.red('❌ Configuration validation failed!\n'));
    console.log(chalk.red(error.message));
    console.log();
    console.log(chalk.yellow('Please check your .env file and ensure all required variables are set.'));
    console.log(chalk.gray('Refer to .env.example for configuration template.\n'));
    process.exit(1);
}
